### PR TITLE
Add 404 "not found" to generated responses

### PR DESCRIPTION
**Pull request by gtanzillo**
Related to https://github.com/ManageIQ/sources-api/pull/95

https://projects.engineering.redhat.com/browse/TPINVTRY-232

### CODE REVIEW

File: lib/manageiq/api/common/open_api/generator.rb [hunk 1]
```diff
@@ -176,6 +176,14 @@ def openapi_list_description(klass_name, primary_collection)
                       "schema" => { "$ref" => build_collection_schema(klass_name) }
                     }
                   }
+                },
+                "404" => {
+                  "description" => "#{klass_name.pluralize} collection",
+                  "content"     => {
+                    "application/json" => {
+                      "schema" => { "$ref" => build_collection_schema(klass_name) }
```
Comment (bdunne) [inline lib/manageiq/api/common/open_api/generator.rb]: This renders a 404 error document
```
{
  "errors": [
    {
      "status": 404,
      "detail": "Record not found"
    }
  ]
}
```

Comment (gtanzillo) [reply lib/manageiq/api/common/open_api/generator.rb]: @bdunne I made this change locally
```diff
diff --git a/lib/manageiq/api/common/open_api/generator.rb b/lib/manageiq/api/common/open_api/generator.rb
index 9efcbda..16528c7 100644
--- a/lib/manageiq/api/common/open_api/generator.rb
+++ b/lib/manageiq/api/common/open_api/generator.rb
@@ -176,16 +176,14 @@ module ManageIQ
                       "schema" => { "$ref" => build_collection_schema(klass_name) }
                     }
                   }
-                },
-                "404" => {
-                  "description" => "#{klass_name.pluralize} collection",
-                  "content"     => {
-                    "application/json" => {
-                      "schema" => { "$ref" => build_collection_schema(klass_name) }
-                    }
-                  },
                 }
-              }
+              },
+              "errors" => [
+                {
+                  "status": 404,
+                  "detail": "Record not found"
+                }
+              ]
```
And that build the 404 like this in the spec -
```
        },
        "errors": [
          {
            "status": 404,
            "detail": "Record not found"
          }
        ]
      },
```
But in other places int he spec we have it this way
```
          },
          "404": {
            "description": "Not found"
          }
        }
      },
```
Which one is correct?

Comment (bdunne) [reply lib/manageiq/api/common/open_api/generator.rb]: The first one is correct, we need the `"errors"` key at the top level of the response.

Comment (gtanzillo) [reply lib/manageiq/api/common/open_api/generator.rb]: Thanks, @bdunne, I'll work on making the necessary changes

File: lib/manageiq/api/common/open_api/generator.rb [hunk 2]
```diff
@@ -176,6 +204,14 @@ def openapi_list_description(klass_name, primary_collection)
                       "schema" => { "$ref" => build_collection_schema(klass_name) }
                     }
                   }
+                },
+                "404" => {
```
Comment (bdunne) [inline lib/manageiq/api/common/open_api/generator.rb]: I think we want to skip this if `primary_collection` is `nil`, right?  We should only have a 404 if there is an id in the path.

**Pull request merged**